### ISSUE TITLE
Scripting does not work correctly

### ISSUE THREAD

**Issue by PerunCreativeMK** [photopea/photopea#3048]
Hi, 
I am trying to automate some tasks using JS in Photophea's Script window but my code works unpredictable. 
For example this code produces two alerts and console output is like: 
\===
Alert: 0
\===
Alert: 1

even when the code works fine in any other compiler - no alert should be invoked.

```
for(let i = 0; i < 2; i++) {
  if (i === 0) continue;
  if (i === 1) continue;

  alert(i);
}
```

I have also a lot of other problems e.g. the assignment is made first and pasting is then made to originalLayers[0] and not to previous active layer. The paste function should be synchronous (at least I didn´t find any information which says something different)
```
docRef.paste(true);
docRef.activeLayer = originalLayers[0];
```

Also when I run code first time it works different than second time e.g. in this example 
```
var docRef = app.activeDocument;
var selection = docRef.selection;
var groups = ["Originals", "Specular", "Normals", "Albedo"];

var originalGroup = docRef.layerSets.getByName("Originals");
var originalLayers = originalGroup.layers;

var shapeRef = [ [0,0], [0,300], [300,300], [300,0] ];
docRef.activeLayer = originalLayers[0]; // <-----

selection.select(shapeRef);
docRef.selection.copy(false);
docRef.activeLayer = docRef.layerSets.getByName("Albedo");
docRef.paste(true);
```
When I run this code it makes copy of part from originalLayers[0] but when I change 0 to 1 the change will not appear until the second re-run of code. Similar problem is when I run this code above in cycle with multiple originalLayers and all the copy and pastes are from one layer even if the alert or console log are writing correct docRef.activeLayer.name in a specific iteration.

I also found out that while cycle is only writing something to console and it does not any iteration.

I do not know, maybe I am doing something wrong but I think there is some problem on your side guys and I am looking forward for some fixes. Thank you.

**Comment by photopea** [comment]
I fixed the problem with "===" and "continue" - they were not implemented in our interpreter.

I fixed the problem with copy-paste, they do not use the OS clipboard now (the browser lets a website access the OS clipboard only asynchronously, the visitor might need to confirm the permission for the website).

Is your code working better now?

**Comment by PerunCreativeMK** [comment]
Hi,
thanks for quick response. Continue and === works fine but I still have problem with copy paste. Now it behaves a little different, my last code from this issue now does not refresh clipboard at all when I change e.g. originalLayers[0] to originalLayers[1]. It refreshes only after I do manual paste to document. Paste to e.g. Paint does not refresh it in Photopea but it paste correct image so select and copy methods are working fine only problem is paste.

Basically what I am trying to do is to make separate textures from one big texture atlas so they can be edited independently . I have atlas and sizes + positions of individual textures in it and copy paste method looked like best soluton for it.

**Comment by photopea** [comment]
Now, a script can paste only the content, which has been copied by the script. It does not access the global clipboard.

Could you share a script, which works differently in Photoshop vs. in Photopea?

**Comment by photopea** [comment]
I have to close it because of lack of details.
